std-bits/base/src/main/java/org/enso/base/Environment_Utils.java

```diff
@@ -1,8 +1,24 @@
 package org.enso.base;
 
 import java.util.HashMap;
+import org.graalvm.polyglot.Context;
 
 public class Environment_Utils {
+  private static final boolean isEspresso;
+
+  static {
+    var name = System.getProperty("java.vm.name");
+    isEspresso = name != null && name.startsWith("Espresso");
+  }
+
+  private Environment_Utils() {}
+
+  public static void safepoint() {
+    if (!isEspresso) {
+      Context.getCurrent().safepoint();
+    }
+  }
```

Comment:
Are you sure this is not affecting our performance?

I always understood that the `Context::safepoint` call was very optimized to be fast. Here we add another layer of branching (`isEspresso`) check + `Context.getCurrent()` on _every_ iteration. I assume the JVM can optimize these relatively well, but I'd like us to make sure.

Reply:
Also, this seems like wrong place to put this. `Environment_Utils` is a helper object for environment variables. It has _nothing_ to do with safepoints. IMO we should move this to a separate file.

Reply:
> Are you sure this is not affecting our performance?

Running [stdlib benchmarks](https://github.com/enso-org/enso/actions/runs/9156252652).

Reply:
> Also, this seems like wrong place to put this. `Environment_Utils` is a helper object for environment variables. It has _nothing_ to do with safepoints. IMO we should move this to a separate file.

I knew this comment was going to come... On the other hand:

The behavior is influenced by `System.getProperty("java.vm.name")` which is kinda an environment (property) variable.

However, as written at https://github.com/enso-org/enso/pull/9997#discussion_r1606481433 - we can't make Espresso work without RegEx and thus we can delay these changes: Reverted in 1be75e6